### PR TITLE
fix(parser): E13 — drop S8.6 in key segments + preserve path-expr whitespace (xx.hocon#42)

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

Implements [xx.hocon E13](https://github.com/o3co/xx.hocon/blob/main/docs/extra-spec-conventions.md#e13) (cross-impl convention, [xx.hocon#42](https://github.com/o3co/xx.hocon/issues/42)) in ts.hocon. Matches Lightbend 1.4.3 behavior for key-position parsing on two axes that ts/rs/go were over-strict on. Pure loosening (no previously-valid input is now rejected); narrow key-string semantics change for inputs with whitespace adjacent to path dots. Target release: **v1.5.3 PATCH** alongside cross-impl rs.hocon / go.hocon PRs.

## What changed

1. **S8.6 not enforced on key path segments** — `foo -bar = 1`, `foo.-bar = 1`, `-foo bar = 1`, `foo -1bar = 1` etc. now accepted verbatim. HOCON.md L270-276 is a value-position lexer rule; key paths are governed by path-element parsing rules.
2. **Path-expression whitespace preserved verbatim around dots** — `a b. c = 1` → `{"a b":{" c":1}}` (leading space on `" c"`); `a b.\tc = 1` → `{"a b":{"\tc":1}}` (HOCON_WS tab uniformly preserved); `a .b = 1` → `{"a ":{"b":1}}` (trailing space on `"a "`).
3. **Bundled correctness fix — trailing-dot key paths now consistently reject** (`foo. = 1`, `a.b. = 1` pre-E13 silently parsed to prefix segments; now throw per Lightbend BadPath).
4. **Bundled correctness fix — dot-WS-dot becomes a WS path segment per Lightbend** (`a. .b = 1` → `["a", " ", "b"]` per `{"a":{" ":{"b":1}}}`; Codex multi-agent-review caught this on the first patch).

## Implementation

- **Lexer** (`src/internal/lexer/`): adds `Token.precedingWhitespace: string` field storing the literal whitespace chars consumed since the previous token. Required so the parser can preserve tab vs space (pw07) and produce the source's actual whitespace verbatim instead of normalising to single ASCII space.
- **Parser** (`src/internal/parser/parser.ts:parseKey`):
  - Removes the S8.6-in-key segment validation loop.
  - Replaces literal `' '` joiner in space-concat path with `t.precedingWhitespace`.
  - Adds `postDotPrefix` state: after a trailing-dot continuation, the next token's `precedingWhitespace` becomes leading prefix of the new segment (E13 path-WS preservation rule).
  - Adds dot-WS-dot branch: postDotPrefix becomes its own segment when next token starts with `.` (Codex P2 fix).
  - Adds post-loop guard: trailing-dot at end of key path throws ParseError (matches Lightbend BadPath).

## Tests

- **New conformance tests** (15 tests): `tests/key-hyphen-position.test.ts` (8 kh* fixtures), `tests/path-expr-whitespace.test.ts` (6 pw* success + 1 pw06 error).
- **New unit tests** for dot-WS-dot and trailing-dot-after-dot-WS-dot (2 tests, Codex review followup).
- **Updated existing tests** (6 tests previously asserted pre-E13 behavior; comments document the prior expectation):
  - `tests/parser.test.ts` — 5 S10.8 tests updated to expect WS-preserving behavior
  - `tests/s8-unquoted-starts.test.ts` — 1 test inverted (`a.-foo = 1` now accepts)
- **Makefile**: adds `KEY_HYPHEN_DIR` and `PATH_EXPR_WS_DIR` to `testdata` target.

All **1098 tests pass** (1090 baseline + 6 updated + 2 dot-WS-dot).

## Multi-agent review

Reviewed by Claude (general-purpose code-reviewer template, Opus) + Codex CLI. Codex flagged P2 dot-WS-dot edge case — empirically verified against Lightbend (typesafe-config 1.4.3 probe) and fixed. Claude flagged 3 minor items — M1 (doc invariant phrasing) and M2 (CHANGELOG omission) addressed; M3 (leading-dot/double-dot silent-accept gap) deferred per scope discipline (no xx.hocon fixture yet, separate follow-up).

## Test plan

- [x] `make testdata` fetches the new key-hyphen-position/ and path-expr-whitespace/ groups
- [x] Conformance tests for kh01–kh08 + pw01–pw07 all pass
- [x] No regressions in 1090 baseline tests; 6 updated tests reflect E13 behavior with audit-trail comments
- [x] Substitution body S8.6 enforcement unchanged (`${-foo}` still rejects)
- [x] Value-position S8.6 / E8 behavior unchanged
- [x] Codex P2 dot-WS-dot fix verified against Lightbend probe
- [ ] CI green (xx.hocon merged; testdata SHA up to date)
- [ ] Cross-impl PRs in rs.hocon + go.hocon to follow this PR

Refs xx.hocon#42

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- **Bundled fix — trailing-dot key paths now consistently reject**. `foo. = 1`, `a.b. = 1`, `a b. = 1`, and `a. . = 1` now throw `ParseError` ("path has a trailing period — empty key segment not allowed"). Pre-E13 these silently parsed to the prefix segments (e.g. `foo. = 1` → `{"foo":1}`). Aligned with Lightbend BadPath and E13 boundary fixture `pw06`. Leading-dot (`.foo = 1`) and double-dot (`a..b = 1`) in key paths are NOT addressed in this PR (pre-existing silent-accept gap, no xx.hocon fixture yet — tracked as a follow-up).
+- **Bundled fix — dot-WS-dot in key paths produces a WS segment per Lightbend**. `a. .b = 1` now yields `["a", " ", "b"]` (was `["a", " b"]` from the initial E13 patch; Lightbend probe confirms `{"a":{" ":{"b":1}}}`). Caught by Codex multi-agent-review on this PR.
+
+#### Implementation
+
+- **Lexer**: `Token.precedingWhitespace: string` field added (the literal whitespace chars consumed since the previous token). `Token.precedingSpace: boolean` retained for clarity at call sites (invariant: `precedingSpace === (precedingWhitespace.length > 0)`).
```
[the remark this answers is not in the document]
Comment (y1o1) [reply CHANGELOG.md]: Fixed in f14170c — replaced the `precedingSpace === (precedingWhitespace.length > 0)` invariant claim with a more accurate description: comments set hadSpace=true without contributing to the whitespace buffer, so the `newline` token after `// foo\n` or `# foo\n` carries `precedingSpace=true` with `precedingWhitespace=""`. The two fields are related but not equivalent. Cross-referenced `src/internal/lexer/token.ts` for the full rationale.

File: src/internal/lexer/token.ts [hunk 2]
```diff
@@ -27,6 +27,21 @@ export type Token = {
...
+  //
+  // Note: precedingSpace may be true while precedingWhitespace is empty when
+  // the token is preceded only by a comment (no literal WS chars). The
+  // boolean is the right signal for concat detection (S10.5 / S10.8); the
+  // string is the right signal for path-WS preservation (E13). They are NOT
+  // strictly equivalent at the contract level — they answer different
+  // questions. The comment-only case cannot fire in practice in current
+  // grammar (comments run to `\n` which emits a newline token), but the
+  // distinction is preserved so future grammar changes do not silently
+  // couple the two signals.
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/internal/lexer/token.ts]: Fixed in f14170c — replaced "cannot fire in practice" with explicit acknowledgement that the comment-only shape DOES occur for the `newline` token emitted after `// foo\n` / `# foo\n` (the only token shape that can carry this state in current grammar). Also clarified that non-newline tokens in concat / path-WS contexts are always either preceded by literal WS chars OR follow a newline that resets the buffer, so consumers of `precedingWhitespace` can rely on the string alone.
